fdbserver/TLogServer.actor.cpp

```diff
@@ -659,33 +659,13 @@ ACTOR Future<Void> updateStorage( TLogData* self ) {
 		if (self->bytesInput - self->bytesDurable >= SERVER_KNOBS->TLOG_SPILL_THRESHOLD) {
 			while(logData->persistentDataDurableVersion != logData->version.get()) {
 				totalSize = 0;
-				std::vector<std::pair<std::deque<std::pair<Version, LengthPrefixedStringRef>>::iterator, std::deque<std::pair<Version, LengthPrefixedStringRef>>::iterator>> iters;
-
-				for(tagLocality = 0; tagLocality < logData->tag_data.size(); tagLocality++) {
-					for(tagId = 0; tagId < logData->tag_data[tagLocality].size(); tagId++) {
-						tagData = logData->tag_data[tagLocality][tagId];
-						if(tagData) {
-							iters.push_back(std::make_pair(tagData->versionMessages.begin(), tagData->versionMessages.end()));
-						}
-					}
-				}
-
-				nextVersion = 0;
-				while( totalSize < SERVER_KNOBS->UPDATE_STORAGE_BYTE_LIMIT || nextVersion <= logData->persistentDataVersion ) {
-					nextVersion = logData->version.get();
-					for( auto &it : iters )
-						if(it.first != it.second)
-							nextVersion = std::min( nextVersion, it.first->first + 1 );
-
-					if(nextVersion == logData->version.get())
-						break;
-
-					for( auto &it : iters ) {
-						while (it.first != it.second && it.first->first < nextVersion) {
-							totalSize += it.first->second.expectedSize();
-							++it.first;
-						}
-					}
+				Map<Version, std::pair<int,int>>::iterator sizeItr = logData->version_sizes.begin();
```

Comment:
I think I get the gist of this, but let's talk it over to make sure I understand it.

Reply:
We talked, and I think I'm satisfied with this.